Source/Things/Acceleration.as

```diff
@@ -0,0 +1,90 @@
+#if !COMPETITION
+class DashboardAcceleration : DashboardThing
+{
+	DashboardAcceleration()
+	{
+	}
+
+	void OnSettingsChanged() override
+	{
+	}
+
+	void RenderAccelerometer(const vec2 &in pos, const vec2 &in size, float acc)
+	{
+		vec2 psize = vec2(size.x, size.y/2);
+		vec2 npos = vec2(pos.x, pos.y);
+		float accHeight = (acc / Setting_Acceleration_MaxAcceleration) * psize.y;
+		
+		nvg::Save();
+		// Negative acc backdrop
+		nvg::BeginPath();
+		nvg::Translate(0, psize.y);
+		nvg::RoundedRect(npos.x, npos.y, psize.x, psize.y, Setting_Acceleration_BorderRadius);
+		nvg::StrokeWidth(Setting_Acceleration_BorderWidth);
+		nvg::StrokeColor(Setting_Acceleration_BorderColor);
+		nvg::FillColor(Setting_Acceleration_BackdropColor);
+		nvg::Fill();
+		
+		if (accHeight < 0) {
+			nvg::Scissor(npos.x, npos.y, psize.x, accHeight * -1.0f);
+			nvg::FillColor(Setting_Acceleration_Negative_Color);
+			nvg::Fill();
+			nvg::ResetScissor();
+		}
+		
+		// Negative acc border
+		nvg::BeginPath();
+		nvg::RoundedRect(npos.x, npos.y, psize.x, psize.y, Setting_Acceleration_BorderRadius);
+		nvg::Stroke();
+
+		// Positive acc backdrop
+		nvg::BeginPath();
+		nvg::Scale(1,-1);
+		nvg::RoundedRect(pos.x, pos.y, psize.x, psize.y, Setting_Acceleration_BorderRadius);
+		nvg::StrokeWidth(Setting_Acceleration_BorderWidth);
+		nvg::StrokeColor(Setting_Acceleration_BorderColor);
+		nvg::FillColor(Setting_Acceleration_BackdropColor);
+		nvg::Fill();
+		
+		if (accHeight >= 0) {
+			nvg::Scissor(pos.x, pos.y, psize.x, accHeight * 1.0f);
+			nvg::FillColor(Setting_Acceleration_Positive_Color);
+			nvg::Fill();
+			nvg::ResetScissor();
+		}
+		
+		// Positive acc border
+		nvg::BeginPath();
+		nvg::RoundedRect(pos.x, pos.y, psize.x, psize.y, Setting_Acceleration_BorderRadius);
+		nvg::Stroke();
+		
+		nvg::Restore();
+
+		if (Setting_Acceleration_ShowTextValue) {
+			float textSize = (m_size.x - Setting_Acceleration_Padding * 3) / 3;
+			nvg::FontSize(Setting_Acceleration_FontSize);
+			nvg::FillColor(Setting_Acceleration_TextColor);
+			nvg::TextBox(textSize, m_size.y / 4, Setting_Acceleration_Padding * 3, Icons::AngleDoubleUp + "\n" + Text::Format("%.2f", acc > 0 ? acc : 0));
+			nvg::TextBox(textSize, m_size.y / 1.33, Setting_Acceleration_Padding * 3, Icons::AngleDoubleDown + "\n" + Text::Format("%.2f", acc < 0 ? Math::Abs(acc) : 0));
+		}
+	}
+
+	void Render(CSceneVehicleVisState@ vis) override
+	{
+		float speed = vis.FrontSpeed;
+		int arr_size = 4;
+		float curr_acc = ((speed - g_prev_speed) / (g_dt/1000));
+		g_prev_speed = speed;
+		g_acc[g_idx] = curr_acc;
+		g_idx = (g_idx + 1) % arr_size;
+		float sum = 0;
+		for(int n = 0; n < arr_size; n++)
+			sum += g_acc[n];
```

Comment:
You could avoid arrays when smoothing using interpolation as well, which is what I'd probably prefer in this case. For example: `value = lerp(value, newValue, 0.9f);` (or some other value than 0.9, but you can play around with it to see what works well)

Additionally, maybe a setting to enable/disable smoothing could be nice.

Reply:
I'll look into using lerp instead.

Reply:
Tested out Math::Lerp and it seems to me that just getting the average over a couple of frames is better.